### PR TITLE
Graph.edgeCount more efficient: NodeDb.outEdgeCount without instantiating edges

### PR DESCRIPTION
**Pull request by mpollmeier**
includes refactorings in separate commit

### CODE REVIEW

File: core/src/main/java/overflowdb/NodeDb.java [hunk 1]
```diff
@@ -35,7 +35,7 @@
   private Object[] adjacentNodesWithProperties = new Object[0];
 
   /* store the start offset and length into the above `adjacentNodesWithProperties` array in an interleaved manner,
-   * i.e. each outgoing edge type has two entries in this array. */
+   * i.e. each adjancent edge type has two entries in this array. */
```
Comment (hubertp) [inline core/src/main/java/overflowdb/NodeDb.java:38]:
```suggestion
   * i.e. each adjacent edge type has two entries in this array. */
```

File: core/src/main/java/overflowdb/NodeDb.java [hunk 2]
```diff
@@ -421,6 +421,26 @@ public void addEdgeSilent(String label, Node inNode, Map<String, Object> keyValu
...
+  protected int outEdgeCount() {
+    int outEdgeCount = 0;
+    for (String label : layoutInformation().allowedOutEdgeLabels()) {
+      int offsetPos = getPositionInEdgeOffsets(Direction.OUT, label);
+      if (offsetPos != -1) {
+        int start = startIndex(offsetPos);
+        int length = blockLength(offsetPos);
+        int strideSize = getStrideSize(label);
+        for (int i = start;
+             i < adjacentNodesWithProperties.length && i < start + length;
```
Comment (hubertp) [inline core/src/main/java/overflowdb/NodeDb.java:433]: compute `start + length` and `adjacentNodesWithProperties.length` only once?

**Review by hubertp — APPROVED** [review]
nits

File: core/src/main/java/overflowdb/NodeDb.java (hunk 2 above)
Comment (mpollmeier) [reply core/src/main/java/overflowdb/NodeDb.java]: `adjacentNodesWithProperties` is an array, so `.length` doesn't compute anything, and JIT will likely even inline it

`start + length`: yup ok will do across the board in this file

**Pull request merged**